sphinx/writers/latex.py

```diff
@@ -1844,37 +1844,43 @@ def visit_figure(self, node):
                 self.body.append('\\capstart')
             self.context.append(ids + '\\end{sphinxfigure-in-table}\\relax\n')
         elif node.get('align', '') in ('left', 'right'):
-            length = None
-            if 'width' in node:
-                length = self.latex_image_length(node['width'])
-            elif 'width' in node[0]:
-                length = self.latex_image_length(node[0]['width'])
-            self.body.append('\\begin{wrapfigure}{%s}{%s}\n\\centering' %
-                             (node['align'] == 'right' and 'r' or 'l', length or '0pt'))
-            self.context.append(ids + '\\end{wrapfigure}\n')
-        elif self.in_minipage:
-            if ('align' not in node.attributes or
-                    node.attributes['align'] == 'center'):
-                self.body.append('\n\\begin{center}')
-                self.context.append('\\end{center}\n')
+            if self.builder.config.latex_use_wrapfig:
+                length = None
+                if 'width' in node:
+                    length = self.latex_image_length(node['width'])
+                elif 'width' in node[0]:
+                    length = self.latex_image_length(node[0]['width'])
+                self.body.append('\\begin{wrapfigure}{%s}{%s}\n\\centering' %
+                                 (node['align'] == 'right' and 'r' or 'l',
+                                  length or '0pt'))
+                self.context.append(ids + '\\end{wrapfigure}\n')
+
+                # emit a warning about wrapfig (refs: #3289)
+                logger.warning(_('A figure requiring left or right alignment '
```

Comment:
Please use double underscore method instead. It's for messaging.
The single one is used for a part of document.

Reply:
Is this better to use info level message like #4202?

Reply:
I think info level indeed is better. I will do it.